app/views/subscriptions_management/index.html.erb

```diff
@@ -91,8 +91,16 @@
       <% end %>
     </span>
 
+    <% heading_text = capture do %>
```

Comment:
humm variable assingment in a view always feels like a smell to me if we can avoid it...
but then one of the branches writes HTML...

@danacotoran, @alex9smith what do you think?

I'm wondering if we move this into the controller and set an instance variable like `@heading_text`, which either contains the `subscription['subscriber_list']['title']` or returns some HTML?

Reply:
Yeah that's fair - I thought the same but then saw that there are loads of other variable assignments in this view and decided to be lazy 😅

Reply:
I didn't want HTML in the controller or a variable in the view. Given the other variables in the view, this felt like the lesser of two evils to me

Reply:
No I think it's fair, I had a similar thought at a glance.

However on closer inspection I think the other variable assignments here are actually doing something kind of cool.
They seem to be using capture blocks to create kind of [mini partials](https://github.com/alphagov/email-alert-frontend/blob/3849d857b1ee4a21908b400398401a2d69f6a90b/app/views/subscriptions_management/index.html.erb#L15-L18) that we can pass cleanly into our components.

I think what's different about what we're doing here is that we're doing an `if` statement within variable assignment

Reply:
I think... then on the basis of complexity we might consider moving this to a helper?

Perhaps we could do something like:
```
def link_or_title(subscription)
  return { <a> ... etc } if is_single_page_subscription?(subscription)

  subscription['subscriber_list']['title']
end
```

Then call:

`link_or_title(subscription)` in the component?

Reply:
though, now I think about it... it's a bit odd to be either returning HTML or a string...
Is that right? We've got a heading that can be a link here?

Let me go have an 👀 at the components

Reply:
Yup, we want to have a heading that's a link if it's a single page subscription and text if not

Reply:
I've had a look around, this seems cool from an accessibility standpoint, we seem to do it everywhere and it renders in the image you've provided, so ignore my momentary  🧠❓

Reply:
I've refactored to use a helper in the controller for the logic. There will probably be a conflict with #1197 now, but it''l be easy enough to resolve.